atlasdb-config/src/main/java/com/palantir/atlasdb/http/RateLimitExceededExceptionMapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.http;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import com.palantir.atlasdb.qos.ratelimit.RateLimitExceededException;
+
+/**
+ * This class maps instances of {@link RateLimitExceededException} to HTTP responses with a 429 status code.
+ * Users may register this exception mapper for a standard way of converting these exceptions to 429 responses, which
+ * are meaningful in that {@link com.palantir.remoting3.jaxrs.JaxRsClient}s are able to back off accordingly.
+ */
+public class RateLimitExceededExceptionMapper implements ExceptionMapper<RateLimitExceededException> {
```

Comment:
Should we register this by default?

Reply:
Main concern is that this writes exceptions in the remoting2 style, but I suspect some internal users will want to write their exceptions in the remoting3 style.

@nziebart any thoughts?

Reply:
i think we've decided it's not atlas' place to modify the API contract of the consumer app, so we shouldn't register by default